app/scripts/metamask-controller.js

```diff
@@ -1347,7 +1347,11 @@ export default class MetamaskController extends EventEmitter {
     this.on('controllerConnectionChanged', (activeControllerConnections) => {
       const { completedOnboarding } =
         this.onboardingController.store.getState();
-      if (activeControllerConnections > 0 && completedOnboarding) {
+      if (
+        activeControllerConnections > 0 &&
+        completedOnboarding &&
+        this.isUnlocked()
```

Comment:
Will this not be a very disruptive change for dapps that use the RPC provider before asking for a users accounts?

Reply:
@FrederikBolding here you're thinking of anything that uses the non protected methods of the RPC provider right? like anything that is asking for general block/network data and not specific to an account. What part of uniswap did you test, @pedronfigueiredo ?

Reply:
What I tested was:
- Connect Uniswap to MetaMask
- Lock the wallet
- When I go back to uniswap and click connect, the pop-up opens asking for the password
- When I enter the password, the wallet is unlocked, and the connection is re-established.

This happens both on my branch and on a prod wallet.

Reply:
> anything that is asking for general block/network data and not specific to an account

Thanks for clarifying @brad-decker! Can you point me to an example / help me understand how it can be tested?

Reply:
> here you're thinking of anything that uses the non protected methods of the RPC provider right?

Yeah, it is currently supported for dapps and Snaps to use read-only RPC methods while the extension is locked. I was worried that this PR would change that, but reading it again it might not actually be changing that. 

If that is what we are trying to change here that seems like a very disruptive change to me and would be a breaking change for the Snaps platform. 

Apologies I don't have a good example of a dapp trying to do this, but it used to be a widely used pattern. Maybe not as much anymore?

Reply:
I followed up with Frederik on Slack, and with his guidance I was able to test that Snaps can still use read-only RPC methods while the extension is locked.